certora/specs/SocialRecoveryModule.spec

```diff
@@ -298,3 +299,27 @@ rule disabledRecoveryModuleResultsInFinalizationRevert(env e) {
             safeContract.getOwners().length == 1 &&
             currentThreshold == safeContract.getThreshold());
 }
+
+// This rule verifies that a guardian can only initiate recovery for the safe account it has been assigned to.
+// Here we only check initiation, and not execution of recovery.
+rule guardiansCanInitiateRecoveryForAssignedAccount(env e, address guardian, address[] newOwners, uint256 newThreshold) {
+    require e.msg.sender == guardian;
+    require e.msg.value == 0;
+    require newOwners.length > 0;
+    require newThreshold > 0 && newThreshold <= newOwners.length;
```

Comment:
This can be proven right? This leads to edges cases in the other PR as well.

Reply:
For this particular function, we cannot, as the `newThreshold` and `newOwners` are only stored after the `executeRecovery(...)`. But I have added an invariant which checks the threshold once recovery executes, following the above rule.